### ISSUE TITLE
erroneous TooManyArguments for number_format

### ISSUE THREAD

**Issue by WDD-Marv** [vimeo/psalm#307]
passing in three arguments to `number_format()` triggers a TooManyArguments error- number_format():

errors with: `number_format($val, 2, '')` on psalm f96f3a467df4f7fd6b333fa097c079266084241d

3v4l triggers warnings about the parameter count being wrong, so perhaps 3 arguments should be "too few arguments" (since the intent might be to `number_format($val, 2, '', ',')` rather than `number_format($val, 2)`.

**Comment by muglug** [comment]
So this is actually very legitimate behaviour.

Psalm allows both 2 and 4 arguments, and its `TooManyArguments` is appropriate if you mean to only pass in two. Presumably if you know of the 4-argument usage, you can just write that code and Psalm will not complain.

**Comment by yacine-truefacet** [comment]
I've noticed a similar issue as well when four arguments are passed to `number_format`.

It can be reproduced by pasting this on the landing page demo:
```php
number_format(10, 2, null, '');
```

Replacing the third `null` parameter with a string makes the error go away. However, by passing `null`, you defer the default value to the function while passing a string will set the new value. I understand why it's being reported but it strikes me as a false flag. I've added `@psalm-suppress` for now.

**Comment by muglug** [comment]
That’s definitely a false positive - I’ll fix

**Comment by teol** [comment]
I still have false positives with number_format()

**Comment by WDD-Marv** [comment]
@muglug still occurring with faae9fd

@teol - similar example as here? https://getpsalm.org/r/e438436e3e

**Comment by teol** [comment]
@WDD-Marv I have to 4 arguments set

> Too many arguments for method number_format - expecting 2 but saw 4
>         return number_format($amount, 2, '.', '');
